socketio/src/asynchronous/client/client.rs

```diff
@@ -87,11 +88,25 @@ impl Client {
             nsp: builder.namespace.to_owned(),
             outstanding_acks: Arc::new(RwLock::new(Vec::new())),
             auth: builder.auth.clone(),
+            data: builder.data.clone().unwrap_or(Arc::new(())),
             builder: Arc::new(RwLock::new(builder)),
             disconnect_reason: Arc::new(RwLock::new(DisconnectReason::default())),
         })
     }
 
+    /// Fetches data given by [`ClientBuilder::data`]
+    pub fn data<D: Send + Sync + 'static>(&self) -> Arc<D> {
+        self.try_data()
+            .expect("Client::data does not match ClientBuilder::data")
+    }
+
+    /// Attempts to fetch data given by [`ClientBuilder::data`]
+    ///
+    /// None is returned if data was not given or data does not match [`ClientBuilder::data`]
+    pub fn try_data<D: Send + Sync + 'static>(&self) -> Option<Arc<D>> {
```

Comment:
Mhm, I don't really like the naming here. Usually a `try_` prefix implies a Result, which we don't have here. How about we name this method `data` (returning an option) and the other one `data_unchecked` which panics? Or we actually don't provide a panic method (because that's usually a bad idea anyways, no library should panic), and just make the upper one return a Result?

Reply:
`data_unchecked` doesn't really sit right with me. Maybe just `try_data` returning the result is better suited.

Reply:
Or rather just having a `custom_data` function that returns the result might be better, not `try_data`